java-frontend/src/main/java/org/sonar/java/se/checks/NullDereferenceCheck.java

```diff
@@ -109,6 +126,27 @@ private static void addDereferenceMessage(List<JavaFileScannerContext.Location>
   }
 
   @Override
+  public void checkEndOfExecutionPath(CheckerContext context, ConstraintManager constraintManager) {
+    ExplodedGraph.Node exitNode = context.getNode();
+    exitNode.parents().stream().forEach(node -> {
+      MethodYield yield = exitNode.selectedMethodYield(node);
+      if (yield instanceof ExceptionalCheckBasedYield && ((ExceptionalCheckBasedYield) yield).raisedByCheck(this)) {
```

Comment:
polymorphism : let's make "raised by check" part of yield interface. (thus changing the name to generatedByCheck(...)) with default implem returning false.